File: tutors/src/main/java/com/popalay/tutors/TutorsBuilder.java
package com.popalay.tutors;

public class TutorsBuilder {

    private int textColorRes;
    private int shadowColorRes;
    private int textSizeRes;
    private int completeIconRes;
    private int spacingRes;
    private int lineWidthRes;
    private boolean cancelable;

    public TutorsBuilder textColorRes(int textColorRes) {
        this.textColorRes = textColorRes;
        return this;
    }

    public TutorsBuilder shadowColorRes(int shadowColorRes) {
        this.shadowColorRes = shadowColorRes;
        return this;
    }

    public TutorsBuilder textSizeRes(int textSizeRes) {
        this.textSizeRes = textSizeRes;
        return this;
    }

    public TutorsBuilder completeIconRes(int completeIconRes) {
        this.completeIconRes = completeIconRes;
        return this;
    }

    public TutorsBuilder spacingRes(int spacingRes) {
        this.spacingRes = spacingRes;
        return this;
    }

    public TutorsBuilder lineWidthRes(int lineWidthRes) {
        this.lineWidthRes = lineWidthRes;
        return this;
    }

    public TutorsBuilder cancelable(boolean cancelable) {
        this.cancelable = cancelable;
        return this;
    }

    public int getTextColorRes() {
        return textColorRes;
    }

    public int getShadowColorRes() {
        return shadowColorRes;
    }

    public int getTextSizeRes() {
        return textSizeRes;
    }

    public int getCompleteIconRes() {
        return completeIconRes;
    }

    public int getSpacingRes() {
        return spacingRes;
    }

    public int getLineWidthRes() {
        return lineWidthRes;
    }

    public boolean isCancelable() {
        return cancelable;
    }

    public Tutors build() {
        return Tutors.newInstance(this);
    }
}

File: tutors-kotlin/src/main/kotlin/com/github/popalay/tutors/TutorsBuilder.kt
package com.github.popalay.tutors

class TutorsBuilder(var textColorRes: Int = 0,
                    var shadowColorRes: Int = 0,
                    var textSizeRes: Int = 0,
                    var completeIconRes: Int = 0,
                    var spacingRes: Int = 0,
                    var lineWidthRes: Int = 0,
                    var cancelable: Boolean = false) {

    constructor(init: TutorsBuilder.() -> Unit) : this() {
        init()
    }

    fun textColorRes(init: TutorsBuilder.() -> Int) = apply { textColorRes = init() }

    fun shadowColorRes(init: TutorsBuilder.() -> Int) = apply { shadowColorRes = init() }

    fun textSizeRes(init: TutorsBuilder.() -> Int) = apply { textSizeRes = init() }

    fun completeIconRes(init: TutorsBuilder.() -> Int) = apply { completeIconRes = init() }

    fun spacingRes(init: TutorsBuilder.() -> Int) = apply { spacingRes = init() }

    fun lineWidthRes(init: TutorsBuilder.() -> Int) = apply { lineWidthRes = init() }

    fun cancelable(init: TutorsBuilder.() -> Boolean) = apply { cancelable = init() }

    fun build() = Tutors.newInstance(this)

}
